descholar-ceo/AnnounceIT
# #170865656 Admin get all announcements from all users api

#### What does this PR ​ do​ ?
As an admin I should be able to get all users announcements

#### Description of Task to be completed?
  * Add unit test for this api
  * Add a function to retrieve all uses' announcements,
 * In the above function to check wether a current user is an admin or not

#### How should this be manually tested?
* Clone this repository in case you don't have it using this command `git clone https://github.com/descholar-ceo/AnnounceIT.git`
* Type these commands : 
    *  `cd AnnounceIT` to go in the root directory of the project
    * `npm install` to install all the packages
    * `touch .env` to create dotenv file which will keep privately sensitive data for you,
    * Open the dotenv created above and copy paste this `PORT=8080` press Enter and copy paste 
     this `TOKEN_KEY=descholar` __you can always modify this data by changing the values of 
     PORT and TOKEN_KEY as you wish,__
    * `npm start` to start the server, sothat the api will be up and running
    * Open [Postman](https://www.getpostman.com/) and
    create a get request to this url `http://localhost:8080/api/v1/announcements/admin-get-all-announcements-from-all-users` as you can see the result shows that you don't have any regiistered announcement yet, [Follow this link to know how to add new announcement](https://github.com/descholar-ceo/AnnounceIT/pull/20)
    * Once you are done with adding new announcement enter again `http://localhost:8080/api/v1/announcements/admin-get-all-announcements-from-all-users`,

#### Any background context you want to provide?
* Use [Postman](https://www.getpostman.com/), to test it or any other rest-api testing tools available to you

#### What are the relevant pivotal tracker stories?
[170865656](https://www.pivotaltracker.com/n/projects/2429498#)

#### Screenshots (​ if​ appropriate)

#### Questions:
No Question
